src/transformers/models/bart/modeling_bart.py

```diff
@@ -321,58 +317,50 @@ def forward(
         # if key_value_states are provided this layer is used as a cross-attention layer
         # for the decoder
         is_cross_attention = key_value_states is not None
-
         bsz, q_len, _ = hidden_states.size()
 
         # get query proj
-        query_states = self._reshape(self.q_proj(hidden_states), -1, bsz)
-        # get key, value proj
-        # `past_key_value[0].shape[2] == key_value_states.shape[1]`
-        # is checking that the `sequence_length` of the `past_key_value` is the same as
-        # the provided `key_value_states` to support prefix tuning
-        if (
-            is_cross_attention
-            and past_key_value is not None
-            and past_key_value[0].shape[2] == key_value_states.shape[1]
-        ):
+        query_states = self.q_proj(hidden_states).view(bsz, -1, self.num_heads, self.head_dim)
```

Comment:
```suggestion
        query_states = self.q_proj(hidden_states).view(bsz, -1, self.num_heads, self.head_dim)
        query_states = query_states * self.scaling
```
to match the eager implementation

Reply:
actually this matches imple before cache and what we had in Llama (before the latest attn refactor), so I think sdpa/fa2 apply scaling based on input shapes automatically (as it is only `head_dim**0.5`)

I'll verify that to be sure

Reply:
Sorry, you are right, it doesn't need it. I mixed it up with another similarly named flag in BART, `scale_embedding` (which can cause subtle differences)

In any case, I see the `self.scale = constant` pattern in many models, we probably should get rid of it in another PR 👼